py/selenium/webdriver/common/bidi/network.py

```diff
@@ -0,0 +1,326 @@
+        response matches the given URL pattern.
+
+        Parameters:
+        ----------
+            callback (function): A function to be executed when url is matched by a url_pattern
+                The callback function receives a `Response` object as its argument.
+            url_pattern (str, optional): A substring to match against the response URL.
+                Default is an empty string, which matches all URLs.
+
+        Returns:
+        -------
+            str: The request ID of the intercepted response.
+        """
+        self.__add_intercept(phases=[self.PHASES["response_started"]])
+
+        def callback_on_url_match(data):
+            # create response object to pass to callback
+            if url_pattern in data["response"]["url"]:
+                request_id = data["request"].get("requestId", uuid.uuid4())
+                url = data["response"].get("url")
+                status_code = data["response"].get("status")
+                body = data["response"].get("body", None)
+                headers = data["response"].get("headers", {})
+                response = Response(request_id, url, status_code, headers, body, self)
+                callback(response)
+
+        request_id = self.__on("response_started", callback_on_url_match)
+        self.callbacks[request_id] = callback
+        if "response_started" not in self.subscriptions or not self.subscriptions.get("response_started"):
+            self.subscriptions["response_started"] = [request_id]
+        else:
+            self.subscriptions["response_started"].append(request_id)
+        return request_id
+
+    def remove_response_handler(self, response_id):
+        """Removes a response handler.
+
+        Parameters:
+        ----------
+            response_id (str): The request ID of the intercepted response.
+        """
+        self.__remove_intercept(request_id=response_id)
+        self.subscriptions["response_started"].remove(response_id)
+        del self.callbacks[response_id]
+        if len(self.subscriptions["response_started"]) == 0:
+            session_unsubscribe(self.conn, self.EVENTS["response_started"])
+
+
+class Request:
+    def __init__(self, request_id, url, method, headers, body, network: Network):
+        self.request_id = request_id
+        self.url = url
+        self.method = method
+        self.headers = headers
+        self.body = body
+        self.network = network
+
+    def continue_request(self):
+        """Continue after sending a request."""
+        params = {"requestId": self.request_id}
+        if self.url is not None:
+            params["url"] = self.url
+        if self.method is not None:
+            params["method"] = self.method
+        if self.headers is not None:
+            params["headers"] = self.headers
+        if self.body is not None:
+            params["body"] = self.body
+        command = {"method": "network.continueRequest", "params": params}
```

Comment:
For a future PR, we might want to have a utility method that builds the commands to make things easier to read